statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,480 @@
+        # different strata are treated as distinct PSUs.  This is the same as                                                                                
+        # the nest option in R.                                                                                                                              
+        if nest:
+            m = max(clust) + 1
+            sclust = clust + m*self.strat
+            _, self.sclust = np.unique(sclust, return_inverse=True)
+        else:
+            self.sclust = clust.copy()
+            
+        # The number of clusters per stratum                                                                                                                 
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.ncs = np.bincount(self.strat[ii])
+
+        # The stratum for each cluster                                                                                                                       
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.sfclust = self.strat[ii]
+
+        # The total number of clusters over all stratum
+        self.nclust = np.sum(self.ncs)
+
+    def summary(self):
+        """
+        The __str__ method for our data
+        """
+        print("Number of observations:", len(self.strat))
+        print("Sum of weights:", self.weights.sum())
+        print("Number of strata:", self.nstrat)
+        print("The number of clusters per stratum:", self.ncs)
+
+
+    def _check_args(self, strata, cluster, weights):
+        """
+        Minor error checking to make sure user supplied any of 
+        strata, cluster, or weights. For unspecified subgroup labels
+        an array of ones is created
+
+        Parameters
+        ----------
+        strata : array-like or None                                                                                                                                 
+            Strata for each observation. If none, an array 
+            of ones is constructed                                                                                                                                 
+        cluster : array-like or None                                                                                                                 
+            Cluster for each observation. If none, an array 
+            of ones is constructed                                                                                                         
+        weights : array-like or None                                                                                                                               
+            The weight for each observation. If none, an array 
+            of ones is constructed
+
+        Returns
+        -------
+        vals : list
```

Comment:
It actually returns three arguments, strata, cluster, weights (not the same as a single list containing these three things)